File: src/main/kotlin/dev/lupluv/simplegamemode/events/PlayerHandler.java
package dev.lupluv.simplegamemode.events;

import dev.lupluv.simplegamemode.utils.UpdateChecker;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;

public class PlayerHandler implements Listener {

    @EventHandler
    public void onJoin(PlayerJoinEvent e){
        if(e.getPlayer().hasPermission("sgm.updates")) UpdateChecker.sendUpdateMessage(e.getPlayer());
    }

}

File: src/main/kotlin/dev/lupluv/simplegamemode/events/PlayerHandler.kt
package dev.lupluv.simplegamemode.events

import dev.lupluv.simplegamemode.utils.UpdateChecker.sendUpdateMessage
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent

class PlayerHandler : Listener {
    @EventHandler
    fun onPlayerJoin(event: PlayerJoinEvent) = with(event) {
        if (player.hasPermission("sgm.updates")) player.sendUpdateMessage()
    }
}
